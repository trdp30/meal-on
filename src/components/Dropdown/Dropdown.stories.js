/**
 *
 * Stories for Dropdown
 *
 *
 */

import React from 'react';
import { Dropdown } from './index';

export default {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: 'Components/New/Dropdown',
  component: Dropdown,
  decorators: [Story => <Story />],
};

const Template = args => <Dropdown {...args} />;

export const Basic = Template.bind({});

Basic.args = {
  placeholder: 'Search Evaluator',
  options: [
    { label: 'Test 0', value: '1' },
    { label: 'Test 1', value: '2' },
    { label: 'Test 3', value: '3' },
  ],
  labelKey: 'label',
  primaryKey: 'value',
};
