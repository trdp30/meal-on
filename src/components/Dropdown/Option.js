import React, { memo } from 'react';
import { components } from 'react-select';

const Option = props => {
  const {
    selectProps: { OptionComponent, primaryKey, labelKey },
  } = props;
  return (
    <components.Option {...props}>
      {OptionComponent ? (
        React.cloneElement(OptionComponent, {
          key: props.data[String(primaryKey)],
          data: props.data,
          primaryKey,
          labelKey,
          ...props,
        })
      ) : (
        <>{props.children}</>
      )}
    </components.Option>
  );
};

export default memo(Option);
