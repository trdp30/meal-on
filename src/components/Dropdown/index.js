/**
 *
 * Dropdown
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Select, { components } from 'react-select';
import { faChevronDown } from '@fortawesome/pro-solid-svg-icons';
import { get, trim } from 'lodash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/pro-light-svg-icons';
import classNames from 'classnames';
import CustomOption from './Option';

function DropdownSearchIndicator(props) {
  if (props.selectProps.isSearchIcon) {
    return (
      <components.DropdownIndicator {...props} className="!p-0">
        <FontAwesomeIcon icon={faSearch} className="text-gray-900" />
      </components.DropdownIndicator>
    );
  }
  return (
    <components.DropdownIndicator {...props}>
      <FontAwesomeIcon icon={faChevronDown} className="text-gray-900 " />
    </components.DropdownIndicator>
  );
}

export const reactSelectDefaultStyles = {
  input: props => ({
    ...props,
    input: {
      boxShadow: 'none !important',
    },
  }),
};

export function Dropdown({
  value,
  isDisabled,
  isLoading,
  isClearable,
  name = 'dropdown',
  options,
  onChange,
  emptyLabel,
  labelKey,
  primaryKey,
  placeholder,
  isSearchable,
  isMulti = false,
  customStyles = {},
  controlShouldRenderValue,
  OptionComponent,
  icon,
  alternateLabelKey,
  customFormatOptionLabel,
  filterOption,
  isSearchIcon = false,
  lightBackground,
}) {
  const noOptionsMessage = () => emptyLabel;

  const formatOptionLabel = option => trim(get(option, labelKey)) || get(option, alternateLabelKey, option);
  const getOptionValue = option => get(option, primaryKey, option);
  return (
    <Select
      value={value}
      isSearchIcon={isSearchIcon}
      onChange={onChange}
      isDisabled={isDisabled}
      isLoading={isLoading}
      isClearable={isClearable}
      isSearchable={isSearchable}
      name={name}
      placeholder={placeholder}
      options={options}
      OptionComponent={OptionComponent}
      getOptionValue={getOptionValue}
      getOptionLabel={formatOptionLabel}
      formatOptionLabel={customFormatOptionLabel || formatOptionLabel}
      noOptionsMessage={noOptionsMessage}
      styles={{
        ...reactSelectDefaultStyles,
        ...customStyles,
      }}
      components={{
        Option: CustomOption,
        DropdownIndicator: DropdownSearchIndicator,
      }}
      classNames={{
        control: ({ isFocused }) =>
          classNames(
            'border rounded px-2 py-[3px] !shadow-0 ',
            isFocused ? '!border-blue-300' : '!border-gray-200',
            lightBackground ? '!bg-white' : '!bg-gray-50',
          ),
        placeholder: () => 'text-sm font-normal leading-[16px] !mx-0 !text-gray-400',
        input: () => '!m-0 !p-0',
        indicatorSeparator: () => 'hidden',
        indicatorsContainer: () => 'h-8',
        singleValue: () => 'text-sm font-normal leading-[16px] !mx-0 !text-gray-900',
        valueContainer: () => '!p-0',
        menuList: () => '!p-2 flex flex-col gap-1.5',
        option: ({ isSelected }) =>
          classNames(
            '!px-1.5 !py-2 !text-sm',
            isSelected ? '!bg-blue-50 !text-blue-700' : 'hover:!bg-gray-50 !bg-white',
          ),
      }}
      isMulti={isMulti}
      controlShouldRenderValue={controlShouldRenderValue}
      menuPlacement="auto"
      data-testid={isDisabled ? 'selectDisabled' : 'selectEnabled'}
      indicatorIcon={icon}
      filterOption={filterOption}
      backspaceRemovesValue={false}
    />
  );
}

Dropdown.propTypes = {
  value: PropTypes.any,
  isDisabled: PropTypes.bool,
  isLoading: PropTypes.bool,
  isClearable: PropTypes.bool,
  name: PropTypes.string,
  options: PropTypes.array,
  emptyLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  labelKey: PropTypes.string,
  primaryKey: PropTypes.string,
  isSearchable: PropTypes.bool,
  placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  onChange: PropTypes.func.isRequired,
  customStyles: PropTypes.object,
  isMulti: PropTypes.bool,
  controlShouldRenderValue: PropTypes.bool,
  alternateLabelKey: PropTypes.string,
  filterOption: PropTypes.func,
  lightBackground: PropTypes.bool,
};

Dropdown.defaultProps = {
  isSearchable: false,
  isMulti: false,
  controlShouldRenderValue: true,
  alternateLabelKey: 'username',
};

export default memo(Dropdown);
