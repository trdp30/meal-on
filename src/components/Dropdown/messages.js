/*
 * Dropdown Messages
 *
 * This contains all the text for the Dropdown component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Dropdown';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the Dropdown component!',
  },
});
