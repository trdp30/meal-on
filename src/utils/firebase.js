// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAnalytics } from "firebase/analytics";
import { getAuth } from "firebase/auth";

const firebaseConfig = {
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_FIREBASE_APP_ID,
  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENT_ID,
};

let app, analytics, auth;

// Initialize Firebase
// export default async function initializeFirebase() {
//   app = await initializeApp(firebaseConfig, "menuOnRoad");
//   analytics = await getAnalytics(app);

//   if (process.env.NODE_ENV !== "production") {
//     // analytics.app.disable();
//   }
//   auth = await getAuth(app);
//   return {
//     app,
//     analytics,
//     auth,
//   };
// }

app = initializeApp(firebaseConfig, "menuOnRoad");
analytics = getAnalytics(app);

if (process.env.NODE_ENV !== "production") {
  // analytics.app.disable();
}
auth = getAuth(app);

console.log("auth", auth);

export { app, analytics, auth };

export default app;
