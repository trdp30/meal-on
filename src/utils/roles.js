const roles = { admin: "ADMIN", owner: "OWNER", manager: "MANAGER" };

export default roles;
