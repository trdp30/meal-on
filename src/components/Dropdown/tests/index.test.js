/**
 *
 * Tests for Dropdown
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'utils/testHelpers';
// import '@testing-library/jest-dom'; // add some helpful assertions

import Dropdown from '../index';

describe('<Dropdown />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const onChange = jest.fn();
    render(<Dropdown onChange={onChange} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it.skip('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const onChange = jest.fn();
    const { container } = render(<Dropdown onChange={onChange} />);
    expect(container).toMatchSnapshot();
  });
});
