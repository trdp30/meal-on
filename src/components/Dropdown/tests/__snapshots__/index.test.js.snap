// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Should render and match the snapshot 1`] = `
<div>
  <div
    class=" css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-3-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="border rounded px-2 py-[3px] !shadow-0  !border-gray-200 !bg-gray-50 css-13cymwt-control"
    >
      <div
        class="!p-0 css-1fdsijx-ValueContainer"
      >
        <div
          class="text-sm font-normal leading-[16px] !mx-0 !text-gray-400 css-1jqq78o-placeholder"
          id="react-select-3-placeholder"
        >
          Select...
        </div>
        <input
          aria-autocomplete="list"
          aria-describedby="react-select-3-placeholder"
          aria-expanded="false"
          aria-haspopup="true"
          aria-readonly="true"
          class="css-mohuvp-dummyInput-DummyInput"
          id="react-select-3-input"
          inputmode="none"
          role="combobox"
          tabindex="0"
          value=""
        />
      </div>
      <div
        class="h-8 css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class="hidden css-1u9des2-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-1xc3v61-indicatorContainer"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-down text-gray-900 "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="dropdown"
      type="hidden"
      value=""
    />
  </div>
</div>
`;
